ste/mgr-JobMgr.go

```diff
@@ -188,10 +198,117 @@ type jobMgr struct {
 
 	// only a single instance of the prompter is needed for all transfers
 	overwritePrompter *overwritePrompter
+
+	initMu    *sync.Mutex
+	initState *jobMgrInitState
 }
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
+var udamSetupOnce = &sync.Once{}
+
+func (jm *jobMgr) setupUserDelegationAuthManager(src string) {
+	// Check if it doesn't already exist.
+	udamSetupOnce.Do(func() {
+		// If the OAuth token is empty, this will fail anyway.
+		// Set an empty UDAM instance that does nothing.
+		if jm.getInMemoryTransitJobState().credentialInfo.OAuthTokenInfo == (common.OAuthTokenInfo{}) {
+			jm.initState.udam = &userDelegationAuthenticationManager{}
+			return
+		}
+
+		srcURL, err := url.Parse(src)
+
+		// no reason for this to fail, ever.
+		common.PanicIfErr(err)
+
+		// Check if the source is pointing at Azure. We shouldn't send the OAuth token anywhere else.
+		// format is account.[blob.core.windows.net], [] is the selection
+		if srcURL.Host[strings.Index(srcURL.Host, ".")+1:] != "blob.core.windows.net" {
```

Comment:
Not sure we can use this.  Since it doesn't work for sovereign clouds. E.g. For Azure China the accounts end in .blob.core.chinacloudapi.cn